src/Rings/mpoly-ideals.jl

```diff
@@ -1884,8 +1884,13 @@ julia> dim(I)
   if I.dim > -1
     return I.dim
   end
+  is_one(I) && return -1 # Catch a boundary case
```

Comment:
Maybe stupid question: if I remember correctly, this is_one call already needs a groebner basis (possibly via singular). Wouldn't it then be more efficient to use this knowledge in Singular.dimension? Or does AlgebraicSolving.dimension make use of that as well?

Reply:
You are right, `is_one()`  computes a GB via Singular. I will work on this tomorrow, there is more that can be optimized, e.g. caching GBs.